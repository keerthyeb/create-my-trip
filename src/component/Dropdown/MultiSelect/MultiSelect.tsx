import {
  FormControl,
  InputLabel,
  MenuItem,
  OutlinedInput,
  Select,
} from "@mui/material";

interface MultiSelectProps {
  options: string[];
  placeHolder: string;
  selectedItems: string[];
  onChangeHandler: any;
}

const MultiSelect: React.FC<MultiSelectProps> = ({
  options,
  placeHolder,
  selectedItems,
  onChangeHandler,
}) => {
  return (
    <FormControl sx={{ m: 0, width: 300 }}>
      <InputLabel>{placeHolder}</InputLabel>
      <Select
        multiple
        value={selectedItems}
        onChange={onChangeHandler}
        input={<OutlinedInput label={placeHolder} />}
      >
        {options.map((option: string) => (
          <MenuItem key={option} value={option}>
            {option}
          </MenuItem>
        ))}
      </Select>
    </FormControl>
  );
};

export default MultiSelect;
