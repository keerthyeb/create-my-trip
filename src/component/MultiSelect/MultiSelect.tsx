import { FormControl, InputLabel, MenuItem, OutlinedInput, Select } from '@mui/material';

const MultiSelect = (props: { options: any; placeHolder: string; selectedItems: any, onChangeHandler: any }) => {
  const { options, placeHolder, selectedItems, onChangeHandler } = props;

  return (
    <FormControl sx={{ m: 0, width: 300 }}>
      <InputLabel>{placeHolder}</InputLabel>
      <Select
        multiple
        value={selectedItems}
        onChange={onChangeHandler}
        input={<OutlinedInput label={placeHolder} />}
      >
        {options.map((option: string) => (
          <MenuItem key={option} value={option}>

            {option}

          </MenuItem>
        ))}
      </Select>
    </FormControl>
  );
}


export default MultiSelect;