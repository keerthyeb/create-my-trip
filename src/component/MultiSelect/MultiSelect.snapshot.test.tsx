import renderer from 'react-test-renderer';
import MultiSelect from './MultiSelect';

describe('MultiSelect Component', () => {
  it('should render MultiSelect component correctly', () => {
    const options = ['Option 1', 'Option 2', 'Option 3'];
    const placeHolder = 'Select options';
    const selectedItems = ['Option 1'];
    const onChangeHandler = jest.fn();

    const component = renderer.create(
      <MultiSelect options={options} placeHolder={placeHolder} selectedItems={selectedItems} onChangeHandler={onChangeHandler} />
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
