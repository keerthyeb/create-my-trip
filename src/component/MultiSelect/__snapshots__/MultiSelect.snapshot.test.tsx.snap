// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MultiSelect Component should render MultiSelect component correctly 1`] = `
<div
  className="MuiFormControl-root css-cquceo-MuiFormControl-root"
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-1jy569b-MuiFormLabel-root-MuiInputLabel-root"
    data-shrink={true}
  >
    Select options
  </label>
  <div
    className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
    onClick={[Function]}
  >
    <div
      aria-controls=":r0:"
      aria-expanded="false"
      aria-haspopup="listbox"
      className="MuiSelect-select MuiSelect-outlined MuiSelect-multiple MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      role="combobox"
      tabIndex={0}
    >
      Option 1
    </div>
    <input
      aria-hidden={true}
      aria-invalid={false}
      className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
      disabled={false}
      onAnimationStart={[Function]}
      onChange={[Function]}
      required={false}
      tabIndex={-1}
      value="Option 1"
    />
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
      data-testid="ArrowDropDownIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
    <fieldset
      aria-hidden={true}
      className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend
        className="css-14lo706"
      >
        <span>
          Select options
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;
